lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt

```diff
@@ -0,0 +1,28 @@
+package com.swmansion.starknet.data.types
+
+sealed class StarknetResult<out T> {
+
+    data class Success<T>(val value: T) : StarknetResult<T>()
+
+    data class Failure<T>(val exception: Exception) : StarknetResult<T>()
+
+    val isSuccessful: Boolean get() = this is Success<T>
```

Comment:
Let's keep method naming the same as in [Kotlin Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/), otherwise it seems a bit confusing to me.